packages/react-reconciler/src/ReactFiberCommitWork.new.js

```diff
@@ -1067,6 +1069,77 @@ function reappearLayoutEffectsOnFiber(node: Fiber) {
   }
 }
 
+function addOrRemovePendingBoundariesOnRoot(
+  finishedRoot: FiberRoot,
+  finishedWork: Fiber,
+) {
+  // This function adds suspense boundaries to the root
+  // or tracing marker's pendingSuspenseBoundaries map.
+  // When a suspense boundary goes from a resolved to a fallback
+  // state we add the boundary to the map, and when it goes from
+  // a fallback to a resolved state, we remove the boundary from
+  // the map.
+
+  // We use stateNode on the Offscreen component as a stable object
+  // that doesnt change from render to render. This way we can
+  // distinguish between different Offscreen instances (vs. the same
+  // Offscreen instance with different fibers)
+  const offscreenInstance = finishedWork.stateNode;
+
+  let prevState: SuspenseState | null = null;
+  if (
+    finishedWork.alternate !== null &&
```

Comment:
Nit: Assign the alternate to a variable, perhaps called `previous` or something. We try to minimize direct access to `alternate` since it's going to be refactored away in the future.